projects/angular/storyteller-tracker/src/app/routes/story/story.component.ts

```diff
@@ -0,0 +1,303 @@
+        this.sharedToggler.setValue(this.story.getIsShared(), { emitEvent: false });
+        return;
+      }
+      this.isStoryUpdating = true;
+      this.story.setIsShared(isShared);
+
+      // Save "Shared / Unshared"
+      this.firebaseService.updateStory(this.story).subscribe(() => {
+        const message: string = this.story.getIsShared() ? 'Shared' : 'Unshared';
+        this.notificationService.success(message);
+        this.sharedToggler.setValue(this.story.getIsShared(), { emitEvent: false });
+        this.isStoryUpdating = false;
+      });
+    });
+  }
+
+  private loadStory(): void {
+    const storyId: string = this.activatedRoute.snapshot.params['id'];
+    this.storySubscription = this.firebaseService.getStory(storyId).subscribe(story => {
+      if (story) {
+        this.story = story;
+        this.sharedToggler.setValue(this.story.getIsShared(), { emitEvent: false });
+        if (!this.isInit) {
+          // Load tracks and moments just once
+          this.isInit = true;
+          this.loadTracks(storyId);
+          this.loadMoments(storyId);
+        }
+      } else {
+        this.router.navigate(['/home']);
+      }
+    });
+  }
+
+  private loadTracks(storyId: string): void {
+    this.tracksSubscription = this.firebaseService.getTracks().subscribe(tracks => {
+      this.tracksSubscription.unsubscribe();
+      this.tracks = tracks.filter(track => track.getStoryId() === storyId);
+      // Display loaded tracks
+      for (const track of this.tracks) {
+        this.tracker.savedMs += track.getEndedMs() - track.getStartedMs();
+      }
+      this.tracker.displayMs(0);
+      this.loadingService.stop();
+    });
+  }
+
+  private loadMoments(storyId: string): void {
+    this.momentsSubscription = this.firebaseService.getMoments().subscribe(moments => {
+      this.momentsSubscription.unsubscribe();
+      this.isMomentsLoading = false;
+      moments = moments.filter(moment => moment.getStoryId() === storyId);
+      moments.sort((a, b) => {
+        // Newest firts
+        return Math.sign(b.getTimestampMs() - a.getTimestampMs());
+      });
+      for (const moment of moments) {
+        const momentUI: MomentUI = {
+          note: moment.getNote(),
+          screenshot: null,
+          timestamp: moment.getTimestampMs(),
+          isSaved: true,
+          isSaving: false,
+          isScreenshotLoading: true,
+        };
+        const index: number = this.momentList.push(momentUI) - 1;
+
+        // We have only relative screenshot url. Getting full link takes some time.
+        if (moment.getScreenshot()) {
+          this.firebaseService.getScreenshotURL(moment.getScreenshot()).subscribe(url => {
+            // Display loading of screenshot image
+            const screenshot = new Image();
+            screenshot.onload = () => {
+              momentUI.isScreenshotLoading = false;
+            };
+            screenshot.src = url;
+            momentUI.screenshot = url;
```

Comment:
You place the `url` in `momentUI.screenshot`, yet in `parseImageBinary()` you use it as a binary.
How does that work? 🤔

Reply:
It's how html `<img src="data" />` works. Data can be both: url and binary.

Reply:
Ok